### DISCUSSION TITLE
Raspberry Pi exemple Subsecond latency.

### DISCUSSION THREAD

**Discussion by gblessas** [OvenMediaLabs/OvenMediaEngine#1239] [Help]
Hi,

i tried your engine for realtime remote camera scenario and it's very interesting.

Do you set up any raspberry pi 4b based camera streaming case.

setting it up is a bit complicated to have efficient streaming, i tested using ffmpeg and rtmp over lte. and webrtc client viewer on pc.

is there a better protocol, solution for high efficient sub second latency with ovenmedia engine ?

I'm looking for best solution before scaling.

**Comment by getroot** [comment]
Both RTMP and SRT are good choices.

**Reply by gblessas** [reply]
Hi, I think it could be a nice promotion tool for ovenmedia to provide an accurate streaming example with low latency (subsecond) streaming based on a raspberry pi.
Actually i use rtmp streaming with libcamera-vid  (in publish to ovenmedia mode) but i'm not sure to have the best configuration for the acquisition many examples exists using gstreamer or ffmpeg but lots of them do not work on bullseye ou buster and have high latencies.
the configuration i managed to set is this one for bullseye :

``
#depending bitrate available on link
HRES=640 
VRES=480
BWIDTH =700000

STREAM_ROTATION=0
FRAMERATE =25
IFRAMES=5

#--sharpness
LIBCAM_CAMERA_SHARPNESS='--sharpness 1.0'
#--contrast
LIBCAM_CAMERA_CONTRAST='--contrast 1.0'
#--brightness,  (-1 1)
LIBCAM_CAMERA_BRIGHTNESS='--brightness 0.0'
#--saturation,
LIBCAM_CAMERA_SATURATION='--saturation 1.0'
#--exposure= normal sport long
LIBCAM_CAMERA_EXPOSURE_PROFILE='--exposure sport'
#--metering= center spot average custom
LIBCAM_CAMERA_METERING='--metering centre'
#--awb= auto incandescent tungsten fluorescent indoor daylight cloudy custom
LIBCAM_CAMERA_AWB='--awb auto'
#--autofocus-mode= default manual auto continuous
LIBCAM_CAMERA_AF='--autofocus-mode default'
LIBCAM_OTHER_OPTIONS='--profile main --level 4.2 --denoise cdn_off'


/bin/libcamera-vid -v 0 -t 0 --nopreview --width HRES --height $VRES--rotation $STREAM_ROTATION --framerate $FRAMERATE -b $BWIDTH -g $IFRAMES  \
    $LIBCAM_CAMERA_SHARPNESS $LIBCAM_CAMERA_CONTRAST $LIBCAM_CAMERA_BRIGHTNESS $LIBCAM_CAMERA_SATURATION \
    $LIBCAM_CAMERA_EXPOSURE_PROFILE $LIBCAM_CAMERA_METERING $LIBCAM_CAMERA_AWB $LIBCAM_CAMERA_AF \
    $LIBCAM_OTHER_OPTIONS \
    --inline \
    --codec libav --libav-format flv -o rtmp://server_url:1935/app/stream_name

``

It could be nice to have SRT/RTMP or WHIP example (at least for me ;) )

**Reply by bchah** [reply]
Your question seems more to do with image acquisition on rPi rather than the live streaming pipeline to OME itself. Is there a particular issue you have related to RTMP or SRT streaming in OME?

As getroot mentioned, both RTMP and SRT will serve you fine here. Maybe use SRT if the server is not on the local network. And for lowest latency playback you should use the WebRTC publisher - HLS will have considerably higher latency.
